src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -253,6 +270,8 @@ func (c controller) UpdateBMHs(wg *sync.WaitGroup) {
 		if allUpdated {
 			c.log.Infof("Updated all the BMH CRs, finished successfully")
 			return
+		} else {
```

Comment:
i don't think this is the right behavior, we don't need set error here

Reply:
i think it will retry here, u can add timeout maybe but we don't need to set error if we don't exit the loop

Reply:
DONE